agent/engine/containermetadata/acquire_metadata.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// acquireNetworkMetadata parses the docker.NetworkSettings struct and
+// packages the desired metadata for JSON marshaling
+func acquireNetworkMetadata(settings *docker.NetworkSettings) (NetworkMetadata, error) {
+	var err error
+	if settings == nil {
+		err = fmt.Errorf("Failed to acquire network metadata: Network metadata not available or does not exist")
+		return NetworkMetadata{}, err
+	}
+
+	// Get Port bindings from docker settings
+	ports, err := api.PortBindingFromDockerPortBinding(settings.Ports)
+	if err != nil {
+		ports = nil
```

Comment:
I think I did this only to formally handle the error given by the function. The reset does nothing because PortBindingFromDockerPortBinding already gives nil for error cases. Is it better to replace the reset with a log documenting this case?